plugin/fulltext/ngram_parser/plugin_ngram.cc

```diff
@@ -68,8 +68,10 @@ ngram_parse(
 		if (next + char_len > end || char_len == 0) {
 			break;
 		} else {
-			/* Skip SPACE */
-			if (char_len == 1 && *next == ' ') {
+			/* Skip SPACE and control characters */
+			int ctype = 0;
+			cs->cset->ctype(cs, &ctype, (uchar*) next, (uchar*) end);
+			if (char_len == 1 && (*next == ' ' || ctype & _MY_CTR)) {
```

Comment:
can you also please check for punctuation characters? Does it crash?

Reply:
What about other parsers?

Reply:
> What about other parsers?

@satya-bodapati, found the same issue with mecab, will take a closer look at it